site/app/libraries/database/DatabaseQueries.php

```diff
@@ -2360,6 +2360,94 @@ public function getAllAnonIds() {
         return $this->course_db->rows();
     }
 
+    /**
+     * Generate notifcation rows
+     *
+     * @param string $source_user_id     user_id of person who generated notification
+     * @param string $type               Type of notification
+     * @param string $metadata           Information about the notification
+     * @param string $content            Text message of notification
+     * @param string $target_users_query Query selection for list of users
+     * @param string $additional_param   Additional parameters to be appended
+     * @param bool   $ignore_self        Should ignore $source_user_id from target users
+     */
+    public function pushNotification($source_user_id, $type, $metadata, $content, $target_users_query, $additional_param, $ignore_self){
```

Comment:
We should have one `pushNotification` function in the `DatabaseQueries` class that takes a `Notification` model instance and switches which operation occurs based on the `Notification`s state.